### DISCUSSION TITLE
blog/react-core-concept-rendering-state

### DISCUSSION THREAD

**Discussion by giscus** [theodorusclarence/theodorusclarence.com#197] [General]
# blog/react-core-concept-rendering-state

Do you wonder why do we use hook instead of normal variable? This will help you redefine some concepts.

https://theodorusclarence.com/blog/react-core-concept-rendering-state

**Comment by theodorusclarence** [comment]
Hello, kindly leave a like (or 5 😄), and comment if you got something to ask! 🚀

**Comment by Allam0053** [comment]
isn't the
```
let count = 0;
```
got the same scope on both quiz 1 and 2 ?

**Reply by theodorusclarence** [reply]
yes they both are the same scope

but on quiz 2, there is a re-render every time it mutates, therefore re-declaring count to 0 each re-render.
